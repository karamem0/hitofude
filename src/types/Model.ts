//
// Copyright (c) 2023-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

export interface AppBarMenuAction {
  type: AppBarMenuType,
  data: unknown
}

export type AppBarMenuType =
  | 'changeTheme';

export interface ContentMenuAction {
  type: ContentMenuType,
  data: unknown
}

export type ContentMenuType =
  | 'closeFile'
  | 'downloadFile'
  | 'editFile'
  | 'openFileVersionPanel'
  | 'saveFile'
  | 'toggleMinimap'
  | 'togglePreview'
  | 'toggleScroll'
  | 'toggleWordWrap';

export interface ContentProps {
  editing?: boolean,
  file?: File,
  loading?: boolean,
  minimap?: boolean,
  preview?: boolean,
  scroll?: boolean,
  scrollPosition?: ScrollPosition,
  text?: string,
  wordWrap?: boolean
}

export interface DialogAction {
  type: DialogType,
  data: unknown
}

export type DialogType =
  | 'changeTheme'
  | 'copyFile'
  | 'copyLink'
  | 'createFile'
  | 'createFolder'
  | 'deleteFile'
  | 'deleteFolder'
  | 'overwriteFile'
  | 'renameFile'
  | 'renameFolder'
  | 'restoreFile';

export interface ExplorerMenuAction {
  type: ExplorerMenuType,
  data: unknown
}

export type ExplorerMenuType =
  | 'copyFile'
  | 'copyLink'
  | 'createFile'
  | 'createFolder'
  | 'deleteFile'
  | 'deleteFolder'
  | 'downloadFile'
  | 'openWithOneDrive'
  | 'refreshFolder'
  | 'renameFile'
  | 'renameFolder'
  | 'toggleAllFiles';

export interface ExplorerParams {
  file?: string,
  folder?: string,
  tab?: TabType.explorer
}

export interface ExplorerProps {
  allFiles?: boolean,
  fileConflicts?: FileConflict[],
  selectedFile?: File,
  selectedFolder?: Folder,
  rootFolder?: Folder
}

export interface File {
  id: string,
  fullName?: string,
  baseName?: string,
  extension?: string,
  mimeType?: string,
  createdDate?: Date,
  updatedDate?: Date,
  webUrl?: string,
  downloadUrl?: string,
  parentId?: string
}

export interface FileConflict {
  id: string,
  name?: string,
  data?: Blob
}

export interface FileVersion {
  id: string,
  version: string,
  updatedDate?: Date,
  size?: number
}

export interface Folder {
  id: string,
  name?: string,
  createdDate?: Date,
  updatedDate?: Date,
  root?: boolean,
  webUrl?: string,
  parentId?: string,
  folders?: Folder[],
  files?: File[]
}

export interface MarkdownProps {
  changed?: boolean,
  defaultText?: string,
  scrollPosition?: ScrollPosition,
  text?: string
}

export type MarkdownToolbarAction =
  | 'bold'
  | 'italic'
  | 'underline';

export type ProgressType =
  | 'save'
  | 'upload';

export type RouteParams = ExplorerParams | SearchParams;

export interface SearchMenuAction {
  type: SearchMenuType,
  data: unknown
}

export type SearchMenuType =
  | 'copyLink'
  | 'openFileLocation';

export interface SearchParams {
  file?: string,
  search?: string,
  tab?: TabType.search
}

export interface SearchProps {
  query?: string,
  resultFiles?: File[],
  selectedFile?: File
}

export interface ScrollPosition {
  scrollX?: number,
  scrollY?: number
}

export interface ScrollSize {
  clientHeight?: number,
  clientWidth?: number,
  scrollHeight?: number,
  scrollWidth?: number
}

export interface SidePanelAction {
  type: SidePanelType,
  data: unknown
}

export type SidePanelType =
  | 'fileVersion';

export interface TabProps {
  loading?: boolean,
  open?: boolean,
  type?: TabType
}

export enum TabType {
  explorer = 0,
  search = 1
}

export enum ThemeName {
  light = 0,
  dark = 1
}
