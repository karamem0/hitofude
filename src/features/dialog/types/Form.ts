//
// Copyright (c) 2023-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

export interface FileCopyDialogFormState {
  baseName?: string
}

export interface FileCreateDialogFormState {
  baseName?: string
}

export interface FileRenameDialogFormState {
  id: string,
  baseName?: string
}

export interface FolderCreateDialogFormState {
  name?: string
}

export interface FolderRenameDialogFormState {
  id: string,
  name?: string
}
