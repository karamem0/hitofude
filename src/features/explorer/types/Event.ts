//
// Copyright (c) 2023-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import { FileRejection } from 'react-dropzone';

export interface DropEventData {
  acceptedFiles: File[],
  rejectedFiles: FileRejection[]
}
