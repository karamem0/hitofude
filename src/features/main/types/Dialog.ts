//
// Copyright (c) 2023 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

export enum DialogType {
  copyFile = 'copyFile',
  createFile = 'createFile',
  createFolder = 'createFolder',
  deleteFile = 'deleteFile',
  deleteFolder = 'deleteFolder',
  renameFile = 'renameFile',
  renameFolder = 'renameFolder'
}

export interface DialogAction {
  type: DialogType,
  payload: unknown
}
