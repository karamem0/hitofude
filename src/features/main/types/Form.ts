//
// Copyright (c) 2023-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

export interface SearchTabItemFormState {
  query?: string
}
