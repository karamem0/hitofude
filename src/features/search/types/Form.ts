//
// Copyright (c) 2023-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

export interface SearchTabPanelFormState {
  query?: string
}

export type SearchTabPanelFormField = keyof SearchTabPanelFormState;
