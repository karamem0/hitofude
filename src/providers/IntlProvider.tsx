//
// Copyright (c) 2023-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import { IntlProvider as Provider, createIntl } from 'react-intl';
import en from '../translations/en.json';
import ja from '../translations/ja.json';

const translations: { [key: string]: Record<string, string> } = {
  en,
  ja
};

const intl = createIntl({
  defaultLocale: 'en',
  locale: window.navigator.language.substring(0, 2),
  messages: translations[window.navigator.language]
});

function IntlProvider(props: Readonly<React.PropsWithChildren<unknown>>) {

  const { children } = props;

  return (
    <Provider {...intl}>
      {children}
    </Provider>
  );

}

export default IntlProvider;
