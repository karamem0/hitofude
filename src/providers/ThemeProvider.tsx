//
// Copyright (c) 2023-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import {
  FluentProvider as Provider,
  Theme
} from '@fluentui/react-components';

import { darkTheme, lightTheme } from '../themes/Theme';
import { InvalidOperationError } from '../types/Error';
import { ThemeName } from '../types/Model';

interface ThemeState {
  theme: Theme,
  themeName: ThemeName
}

interface ThemeContextState {
  theme: ThemeState,
  changeTheme: (themeName: ThemeName) => void
}

const ThemeContext = React.createContext<ThemeContextState | undefined>(undefined);

export const useTheme = (): ThemeContextState => {
  const value = React.useContext(ThemeContext);
  if (value == null) {
    throw new InvalidOperationError();
  }
  return value;
};

function ThemeProvider(props: Readonly<React.PropsWithChildren<unknown>>) {

  const { children } = props;

  const [ theme, setTheme ] = React.useState<ThemeState>({
    theme: lightTheme,
    themeName: ThemeName.light
  });

  const handleChangeTheme = React.useMemo(() => (value: ThemeName) => {
    switch (value) {
      case ThemeName.light:
        setTheme(() => ({
          theme: lightTheme,
          themeName: value
        }));
        break;
      case ThemeName.dark:
        setTheme(() => ({
          theme: darkTheme,
          themeName: value
        }));
        break;
      default:
        break;
    }
  }, []);

  const value = React.useMemo<ThemeContextState>(() => ({
    theme,
    changeTheme: handleChangeTheme
  }), [
    theme,
    handleChangeTheme
  ]);

  return (
    <ThemeContext.Provider value={value}>
      <Provider theme={value.theme.theme}>
        {children}
      </Provider>
    </ThemeContext.Provider>
  );

}

export default ThemeProvider;
