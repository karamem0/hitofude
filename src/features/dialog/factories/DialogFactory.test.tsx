//
// Copyright (c) 2023-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { DialogType } from '../../../types/Model';
import IntlProvider from '../../../providers/IntlProvider';
import Presenter from './DialogFactory.presenter';
import ThemeProvider from '../../../providers/ThemeProvider';

jest.mock('../components/FileCopyDialog', () =>
  function FileCopyDialog({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="test-FileCopyDialog">
        {children}
      </div>
    );
  });

jest.mock('../components/FileCreateDialog', () =>
  function FileCreateDialog({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="test-FileCreateDialog">
        {children}
      </div>
    );
  });

jest.mock('../components/FileDeleteDialog', () =>
  function FileDeleteDialog({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="test-FileDeleteDialog">
        {children}
      </div>
    );
  });

jest.mock('../components/FileOverwriteDialog', () =>
  function FileOverwriteDialog({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="test-FileOverwriteDialog">
        {children}
      </div>
    );
  });

jest.mock('../components/FileRenameDialog', () =>
  function FileRenameDialog({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="test-FileRenameDialog">
        {children}
      </div>
    );
  });

jest.mock('../components/FileRestoreDialog', () =>
  function FileRestoreDialog({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="test-FileRestoreDialog">
        {children}
      </div>
    );
  });

jest.mock('../components/FolderCreateDialog', () =>
  function FolderCreateDialog({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="test-FolderCreateDialog">
        {children}
      </div>
    );
  });

jest.mock('../components/FolderDeleteDialog', () =>
  function FolderDeleteDialog({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="test-FolderDeleteDialog">
        {children}
      </div>
    );
  });

jest.mock('../components/FolderRenameDialog', () =>
  function FolderRenameDialog({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="test-FolderRenameDialog">
        {children}
      </div>
    );
  });

jest.mock('../components/LinkCopyDialog', () =>
  function LinkCopyDialog({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="test-LinkCopyDialog">
        {children}
      </div>
    );
  });

jest.mock('../components/ThemeChangeDialog', () =>
  function ThemeChangeDialog({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="test-ThemeChangeDialog">
        {children}
      </div>
    );
  });

it('should create a shapshot when the type parameter is changeTheme', () => {
  const params = {
    action: {
      type: 'changeTheme' as DialogType,
      data: undefined
    }
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByTestId('test-ThemeChangeDialog')).toBeInTheDocument();
});

it('should create a shapshot when the type parameter is copyFile', () => {
  const params = {
    action: {
      type: 'copyFile' as DialogType,
      data: undefined
    }
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByTestId('test-FileCopyDialog')).toBeInTheDocument();
});

it('should create a shapshot when the type parameter is copyLink', () => {
  const params = {
    action: {
      type: 'copyLink' as DialogType,
      data: undefined
    }
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByTestId('test-LinkCopyDialog')).toBeInTheDocument();
});

it('should create a shapshot when the type parameter is createFile', () => {
  const params = {
    action: {
      type: 'createFile' as DialogType,
      data: undefined
    }
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByTestId('test-FileCreateDialog')).toBeInTheDocument();
});

it('should create a shapshot when the type parameter is createFolder', () => {
  const params = {
    action: {
      type: 'createFolder' as DialogType,
      data: undefined
    }
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByTestId('test-FolderCreateDialog')).toBeInTheDocument();
});

it('should create a shapshot when the type parameter is deleteFile', () => {
  const params = {
    action: {
      type: 'deleteFile' as DialogType,
      data: undefined
    }
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByTestId('test-FileDeleteDialog')).toBeInTheDocument();
});

it('should create a shapshot when the type parameter is deleteFolder', () => {
  const params = {
    action: {
      type: 'deleteFolder' as DialogType,
      data: undefined
    }
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByTestId('test-FolderDeleteDialog')).toBeInTheDocument();
});

it('should create a shapshot when the type parameter is overwriteFile', () => {
  const params = {
    action: {
      type: 'overwriteFile' as DialogType,
      data: undefined
    }
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByTestId('test-FileOverwriteDialog')).toBeInTheDocument();
});

it('should create a shapshot when the type parameter is renameFile', () => {
  const params = {
    action: {
      type: 'renameFile' as DialogType,
      data: undefined
    }
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByTestId('test-FileRenameDialog')).toBeInTheDocument();
});

it('should create a shapshot when the type parameter is restoreFile', () => {
  const params = {
    action: {
      type: 'restoreFile' as DialogType,
      data: undefined
    }
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByTestId('test-FileRestoreDialog')).toBeInTheDocument();
});

it('should create a shapshot when the type parameter is renameFolder', () => {
  const params = {
    action: {
      type: 'renameFolder' as DialogType,
      data: undefined
    }
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByTestId('test-FolderRenameDialog')).toBeInTheDocument();
});

it('should create a shapshot when the type parameter is undefined', () => {
  const params = {
    action: undefined
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
