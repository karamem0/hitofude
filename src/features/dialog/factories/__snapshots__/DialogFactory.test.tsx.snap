// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create a shapshot when the type parameter is changeTheme 1`] = `
<DocumentFragment>
  <div
    class="fui-FluentProvider fui-FluentProviderr0 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      data-testid="test-ThemeChangeDialog"
    />
  </div>
</DocumentFragment>
`;

exports[`should create a shapshot when the type parameter is copyFile 1`] = `
<DocumentFragment>
  <div
    class="fui-FluentProvider fui-FluentProviderr1 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      data-testid="test-FileCopyDialog"
    />
  </div>
</DocumentFragment>
`;

exports[`should create a shapshot when the type parameter is copyLink 1`] = `
<DocumentFragment>
  <div
    class="fui-FluentProvider fui-FluentProviderr2 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      data-testid="test-LinkCopyDialog"
    />
  </div>
</DocumentFragment>
`;

exports[`should create a shapshot when the type parameter is createFile 1`] = `
<DocumentFragment>
  <div
    class="fui-FluentProvider fui-FluentProviderr3 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      data-testid="test-FileCreateDialog"
    />
  </div>
</DocumentFragment>
`;

exports[`should create a shapshot when the type parameter is createFolder 1`] = `
<DocumentFragment>
  <div
    class="fui-FluentProvider fui-FluentProviderr4 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      data-testid="test-FolderCreateDialog"
    />
  </div>
</DocumentFragment>
`;

exports[`should create a shapshot when the type parameter is deleteFile 1`] = `
<DocumentFragment>
  <div
    class="fui-FluentProvider fui-FluentProviderr5 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      data-testid="test-FileDeleteDialog"
    />
  </div>
</DocumentFragment>
`;

exports[`should create a shapshot when the type parameter is deleteFolder 1`] = `
<DocumentFragment>
  <div
    class="fui-FluentProvider fui-FluentProviderr6 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      data-testid="test-FolderDeleteDialog"
    />
  </div>
</DocumentFragment>
`;

exports[`should create a shapshot when the type parameter is overwriteFile 1`] = `
<DocumentFragment>
  <div
    class="fui-FluentProvider fui-FluentProviderr7 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      data-testid="test-FileOverwriteDialog"
    />
  </div>
</DocumentFragment>
`;

exports[`should create a shapshot when the type parameter is renameFile 1`] = `
<DocumentFragment>
  <div
    class="fui-FluentProvider fui-FluentProviderr8 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      data-testid="test-FileRenameDialog"
    />
  </div>
</DocumentFragment>
`;

exports[`should create a shapshot when the type parameter is renameFolder 1`] = `
<DocumentFragment>
  <div
    class="fui-FluentProvider fui-FluentProviderra ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      data-testid="test-FolderRenameDialog"
    />
  </div>
</DocumentFragment>
`;

exports[`should create a shapshot when the type parameter is restoreFile 1`] = `
<DocumentFragment>
  <div
    class="fui-FluentProvider fui-FluentProviderr9 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      data-testid="test-FileRestoreDialog"
    />
  </div>
</DocumentFragment>
`;

exports[`should create a shapshot when the type parameter is undefined 1`] = `
<DocumentFragment>
  <div
    class="fui-FluentProvider fui-FluentProviderrb ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  />
</DocumentFragment>
`;
